pkg/vsphere/disk/disk_manager.go

```diff
@@ -223,6 +223,20 @@ func (m *Manager) Create(ctx context.Context, newDiskURI string,
 //	return nil
 // }
 
+func (m *Manager) Attach(ctx context.Context, spec *types.VirtualDisk) error {
+	machineSpec := configureDeviceSpec(ctx, *m.vm, types.VirtualDeviceConfigSpecOperationAdd, types.VirtualDeviceConfigSpecFileOperationCreate, spec)
+
+	_, err := tasks.WaitAndRetryForResult(ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
```

Comment:
If reconfigure is the operation which needs to retry, then shouldnt the vsphere pkg in vic implement the reconfigure operation (if it doesnt already) and use the retry there?